### ISSUE TITLE
Build failed with 6.0.4 GA

### ISSUE THREAD

**Issue by quentinkb** [hansemannn/titanium-google-maps#89]
I have this error when i try to build my project with 6.0.4 GA once i've included the module. 

This is what i get building directly on my iPhone (5s)
```
[ERROR] :  ** BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :  	Ld build/Intermediates/LPA.build/Debug-iphoneos/LPA.build/Objects-normal/armv7/LPA normal armv7
[ERROR] :  	Ld build/Intermediates/LPA.build/Debug-iphoneos/LPA.build/Objects-normal/arm64/LPA normal arm64
[ERROR] :  (2 failures)
```

This is what i get building on the simulator : 
```
[ERROR] :  ** BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :      Ld build/Products/Debug-iphonesimulator/LPA.app/LPA normal x86_64
[ERROR] :  (1 failure)
```

I have included the module (3.4.0) using appcelerator IDE.

**Comment by yozef** [comment]
Hmmm, I haven't stumbled on this... currently using Ti SDK 6.0.4.GA with ti.googlemaps 3.4.0

**Comment by quentinkb** [comment]
I tried on another computer on another project and got the same issue.. i'm using macOS Sierra 10.12.4

**Comment by yozef** [comment]
Did it work prior to 6.0.4.GA?

Have you checked in `run-on-mainthread` is set to true in tiapp.xml?

**Comment by quentinkb** [comment]
First time i try to use this module, so i can't tell. i've got no entry called "run-on-mainthread" in my tiapp.xml

**Comment by yozef** [comment]
there you have it... make sure you add this line in tiapp.xml:
```
<ios>
        <!-- Add this Line Below -->
        <run-on-mainthread>true</run-on-mainthread>

        <enable-launch-screen-storyboard>false</enable-launch-screen-storyboard>
        <plist>
            <dict>

```

**Comment by quentinkb** [comment]
just edited my comment above, i currently have this line: 
    <property name="run-on-main-thread" type="bool">true</property>

**Comment by yozef** [comment]
please paste the line as indicated inside <ios> tag.

**Comment by quentinkb** [comment]
i did so, 
```
  <ios>
    	<run-on-mainthread>true</run-on-mainthread>
        <enable-launch-screen-storyboard>true</enable-launch-screen-storyboard>
        <use-app-thinning>true</use-app-thinning>
```
still get this error:
```
[ERROR] :  ** BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :  	Ld build/Products/Debug-iphonesimulator/LPA.app/LPA normal x86_64
[ERROR] :  (1 failure)
```

btw, if i take a look into the directory : build/Products/Debug-iphonesimulator/LPA.app/
the "LPA" file does not exist

**Comment by quentinkb** [comment]
If i build with : 
`titanium build --platform ios --build-only`
i get this error : 
```
[ERROR] : Cannot find module '/Users/copark/Documents/Appcelerator_Studio_Workspace/copark/undefined'
[DEBUG]     at Function.Module._resolveFilename (module.js:325:15)
[DEBUG]     at Function.Module._load (module.js:276:25)
[DEBUG]     at Function.Module.runMain (module.js:441:10)
[DEBUG]     at startup (node.js:139:18)
[DEBUG]     at node.js:974:3
[ERROR] Alloy compiler failed
```

**Comment by quentinkb** [comment]
btw, if i try to build the project myself, this is the result: 

```
[INFO]  Project directory: /Users/copark/Downloads/titanium-google-maps-master/ios
[INFO]  Module ID: ti.googlemaps
[INFO]  Creating universal library
[INFO]  Verifying universal library
[INFO]  Creating module zip
[INFO]  Writing module zip: /Users/copark/Downloads/titanium-google-maps-master/ios/ti.googlemaps-iphone-3.4.0.zip
2017-05-23T15:15:36.401Z | ERROR  | An uncaught exception was thrown!
undefined
2017-05-23T15:15:36.402Z | ERROR  | undefined
iMac-de-COPARK:ios copark$ 
```

**Comment by hansemannn** [comment]
Don't use `<run-on-mainthread>true</run-on-mainthread>` in `<ios/>`, use `<property name="run-on-main-thread" type="bool">true</property>` on the top-level. Please also try ti.googlemaps 3.3.x instead of 3.4.0, maybe there was a packaging error there.

**Comment by quentinkb** [comment]
Got the exact same issue with version 3.3.1 but compiling with 3.0.0

**Comment by appsgit1** [comment]
To fix this issue, Add the ti.googlemap module inside your project module folder. Not in the global. This is working fine.

**Comment by quentinkb** [comment]
Where is that folder exactly ? i mean the path to add module locally ?

**Comment by quentinkb** [comment]
Installing the module locally actually works. for version 3.4.0 !

**Comment by hansemannn** [comment]
Very very interesting. Let me try to reproduce with a global installation, I also usually test with local references.

**Comment by hansemannn** [comment]
Fixed it, PR coming asap.

**Comment by hansemannn** [comment]
Fix-version determined to 3.4.1, release today. Thanks guys!
